nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -77,10 +79,12 @@ def _ml_parametrization(self,
 
         return result / self.num_data  # We return a 10-fold validation error.
 
-    def __init__(self, regressor: str, data_dimension: int):
+    def __init__(self, regressor: str, data_dimension: tp.Optional[int] = None, dataset_name: str = "artificial"):
```

Comment:
at may be simpler to just have `dataset: tp.Union[str, int]`: either it's a int and it's artificial, or it's not and it's the given dataset. That avoids coupled parameters

Reply:
This means that we have a single artificial dataset ?
I would go for having dozens of artificial cases :-)

Reply:
if you think you will have more, ok ;)
but then it's a bit weird that your default parameters make the function crash since data_dimension cannot be 0 for the artificial dataset
Also, `dataset` may be enough instead of `dataset_name`